### PR TITLE
Consolidated save_pretrained save path creation into one

### LINKED ISSUE #216: Refactor model path generation into one function out of HuggingFaceHubBucketFSModelTransferSP and LoadLocalModel
Currently the model path is created in different places. this is confusing and error prone. we want to pull the creation of the model path into a seperate function and use the new function in HuggingFaceHubBucketFSModelTransferSP and LoadLocalModel

### PR DESCRIPTION
**Pull request by MarleneKress79789**
…function

Fixes #216 

### All Submissions:

* [x] Is the title of the Pull Request correct?
* [x] Is the title of the corresponding issue correct?
* [x] Have you updated the changelog?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../../pulls) for the same update/change?  
* [x] Are you mentioning the issue which this PullRequest fixes ("Fixes...")
* [x] Before you merge don't forget to run tests in AWS CodeBuild, by adding `[CodeBuild]` to the commit message
*

### CODE REVIEW

File: doc/user_guide/user_guide.md [hunk 1]
```diff
@@ -358,7 +358,7 @@ You can download the model using python like this:
         # download the model and tokenizer from Hugging Face
         model = model_factory.from_pretrained(model_name)
         # save the downloaded model using the save_pretrained function
-        model_save_path = <save_path> / "pretrained" / <model_name>
+        model_save_path = <save_path> / "pretrained" / <model_name> #todo do we use the path function here? or should this be using none of our functions?
```
Comment (tkilias) [inline doc/user_guide/user_guide.md:361]: To be honest, we probably need to get rid of this, the user shouldn't need to handle any of the weird paths

File: tests/integration_tests/with_db/test_upload_model.py [hunk 2]
```diff
@@ -30,8 +30,8 @@ def test_model_upload(setup_database, pyexasol_connection, tmp_path: Path,
                       bucketfs_location: BucketFSLocation, bucketfs_config: config.BucketFs):
     sub_dir = 'sub_dir'
     model_name = model_params.base_model
-    download_path = download_model(model_name, tmp_path)
-    upload_path = bucketfs_operations.get_model_path_with_pretrained(
+    download_path = download_model_to_standard_local_save_path(model_name, tmp_path)
```
[the remark this answers is not in the document]
Comment (MarleneKress79789) [reply tests/integration_tests/with_db/test_upload_model.py:33]: it is be possible to use any path for the model download path here, as it is only the path used to store the model locally before then uploading it to the bucketfs. we could add a test that checks with another local model path to make sure it stays working with any path if you want.

File: doc/user_guide/user_guide.md (hunk 1 above)
Comment (MarleneKress79789) [reply doc/user_guide/user_guide.md]: changed in docu. path does not need to be specified by us here

File: tests/integration_tests/with_db/test_upload_model.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (tkilias) [reply tests/integration_tests/with_db/test_upload_model.py:33]: yes, makes sense, but probably in a different PR. Please, create a ticket.

**Review by tkilias — APPROVED** [review]

**Pull request merged**